controllers/past-grants/index.js

```diff
@@ -55,72 +83,94 @@ router.use(sMaxAge('1d'), injectBreadcrumbs, (req, res, next) => {
     next();
 });
 
-router.get('/', async (req, res) => {
-    /**
-     * Pick out an allowed list of query parameters for forward on to the grants API
-     * @type {object}
-     */
-    const facetParams = pick(req.query, ['q', 'amount', 'postcode', 'programme', 'year', 'orgType', 'sort']);
-    const queryWithPage = Object.assign({}, facetParams, { page: req.query.page || 1 });
-
-    const data = await request({
-        url: PAST_GRANTS_API_URI,
-        json: true,
-        qs: queryWithPage
-    });
-
-    const commonSortOptions = [
-        {
-            label: 'Oldest first',
-            value: 'awardDate|desc'
-        },
-        {
-            label: 'Lowest amount first',
-            value: 'amountAwarded|asc'
-        },
-        {
-            label: 'Highest amount first',
-            value: 'amountAwarded|desc'
+router
+    .route('/')
+    .post(async (req, res) => {
+        // @TODO how can this handle page numbers / sort options?
+        // Do we need hidden input fields?
+        const queryWithPage = addPaginationParameters(buildAllowedParams(req.body), req.body.page);
+        const grantData = await queryGrantsApi(queryWithPage);
+
+        // Repopulate existing app globals so Nunjucks can read them
+        // outside of Express's view engine context
+        const context = Object.assign({}, res.locals, { grants: grantData.results });
```

Comment:
I'm actually fairly pleased with this solution – it was complaining before I put this in about some global functions (eg. `formatDate`). This way, we won't have silently-failing AJAX calls if we use a new global in the template or one of the macros it includes and forget to pass it in specifically.